projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.ts

```diff
@@ -0,0 +1,205 @@
+      tap(
+        schema => {
+          if (schema != null) {
+            this.fields = [
+              this._formlyJsonschema.toFieldConfig(orderedJsonSchema(schema.schema), {
+                // post process JSONSChema7 to FormlyFieldConfig conversion
+                map: (field: FormlyFieldConfig, jsonSchema: JSONSchema7) => {
+                  if (field.templateOptions.label === 'Email') {
+                    if (field.asyncValidators == null) {
+                      field.asyncValidators = {};
+                    }
+                    field.asyncValidators.uniqueEmailMessage = this.getUniqueValidator('email');
+                  }
+                  if (field.templateOptions.label === 'Username') {
+                    if (field.asyncValidators == null) {
+                      field.asyncValidators = {};
+                    }
+                    field.asyncValidators.uniqueUsernameMessage = this.getUniqueValidator('username');
+                  }
+                  return field;
+                }
+              })
+            ];
+          }
+        }
+      ),
+      switchMap(() => {
+        if (this.userID == null) {
+          return of({});
+        }
+        return this._recordService.getRecord('users', this.userID);
+      }),
+      map(user => user.metadata)
+    ).subscribe(model => this.model = model);
+  }
+
+  /**
+   * Retrieve a User given an email or a username.
+   *
+   * @param query - username or email to retrieve a User
+   */
+  searchValueUpdated(query: (string | null)): void {
+    this._recordService.getRecords('users', query).pipe(
+      map((res: any) => {
+        if (res.hits.hits.length === 0) {
+          this._toastService.warning(
+            this._translateService.instant('User not found.')
+          );
+          return of(null);
+        }
+
+        return res.hits.hits[0];
+      }),
+      map(model => {
+        const roles = model.metadata.roles as Array<string>;
+        if (roles) {
+          const alreadyExists = roles.some(v => ['patron', 'librarian', 'system_librarian'].some(r => r === v));
+          if (alreadyExists) {
+            this._toastService.info(
+              this._translateService.instant('This person is already registered in your organisation.')
+            );
+            return of(null);
+          }
+        }
+        this._toastService.info(
+          this._translateService.instant('The personal data has been successfully linked to this patron.')
+        );
+        this.userID = model.id;
+        return this.model = model.metadata ? model.metadata : null;
+      }),
+    ).subscribe();
+  }
+
+  /**
+   * Submit the data if the form is valid.
+   *
+   * Create if the userID is null else update.
+   */
+  submit(): void {
+    this.form.updateValueAndValidity();
+    if (this.form.valid === false) {
+      this._toastService.error(
+        this._translateService.instant('The form contains errors.')
+      );
+      return;
```

Comment:
why ? This function return `void`

Reply:
@Garfield-fr the return here is to not continue the function process